### PR TITLE
Make vof dcdd optional

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
### Description

It was noticed that using the discontinuity-capturing directional dissipation (DCDD) stabilization was not necessarily helpful for all VOF simulations. With this PR, the stabilization term is now optional. It can be enabled/disabled with the help of a boolean parameter (`vof dcdd stabilization`) in the `stabilization` subsection.

### Testing

All failing tests because of the disabled DCDD stabilization term were updated and a new application test (`gls_droplet_marangoni_effect_with_dcdd_stabilization`) has been added with the stabilization enabled.

### Documentation

Stabilization documentation has been updated.

### Checklist (will be removed when merged)
See [this page](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#pull-requests) for more information about the pull request process.

Code related list:
- [x] All in-code documentation related to this PR is up to date (Doxygen format)
- [x] Lethe documentation is up to date
- [x] The branch is rebased onto master
- [x] Code is indented with indent-all and .prm files (examples and tests) with prm-indent
- [x] If parameters are modified, the tests and the documentation of examples are up to date
- [x] Changelog (CHANGELOG.md) is up to date if the refactor affects the user experience or the codebase

Pull request related list:
- [x] No other PR is open related to this refactoring
- [x] Labels are applied
- [x] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [x] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [x] If any future works is planed, an issue is opened
- [ ] The PR description is cleaned and ready for merge

### CODE REVIEW

File: source/core/parameters.cc [hunk 1]
```diff
@@ -712,7 +712,15 @@ namespace Parameters
...
-        "Apply discontinuity-capturing directional dissipation stabilization term on heat transfer <true|false>");
+        "Apply discontinuity-capturing directional dissipation stabilization "
+        "term on heat transfer <true|false>");
+
+      prm.declare_entry(
+        "vof dcdd stabilization",
+        "false",
+        Patterns::Bool(),
+        "Apply discontinuity-capturing directional dissipation (dcdd) "
+        "stabilization term on the VOF phase fraction <true|false>");
```
Comment (blaisb) [inline source/core/parameters.cc:723]: In general the DCDD stabilization still leads to good result. I do not think that it should be disabled by default. We have good results for pretty much except the marangoni droplet using it.

File: source/core/parameters.cc [hunk 2]
```diff
@@ -742,7 +750,9 @@ namespace Parameters
       else
         throw(std::runtime_error("Invalid stabilization strategy"));
 
+      // DCDD stabilization activation parameters
       heat_transfer_stabilization = prm.get_bool("heat transfer stabilization");
```
Comment (blaisb) [inline source/core/parameters.cc:754]: This parameter could also be then heat transfer dcdd stabilization

**Review by blaisb — CHANGES_REQUESTED** [review]
I am not sure it's a good idea to change the default behavior to no dcdd. Let's keep it enabled (it's been enabled for a year now) and until we figure out when it's good and when it's bad.

File: include/solvers/vof_assemblers.h [hunk 3]
```diff
@@ -154,4 +154,60 @@ class VOFAssemblerBDF : public VOFAssemblerBase<dim>
   std::shared_ptr<SimulationControl> simulation_control;
 };
 
+
+/**
+ * @brief Assemble the discontinuity-capturing directional dissipation
+ * stabilization term for the VOF phase fraction.
```
Comment (hepap) [inline include/solvers/vof_assemblers.h:160]:
```suggestion
 * @brief Assemble the Discontinuity-Capturing Directional Dissipation
 * (DCCDD) stabilization term for the VOF phase fraction.
```

File: include/solvers/vof_assemblers.h [hunk 4]
```diff
@@ -154,4 +154,60 @@ class VOFAssemblerBDF : public VOFAssemblerBase<dim>
...
+ *
+ * @note For more information see Tezduyar, T. E. (2003). Computation of
+ * moving boundaries and interfaces and stabilization parameters. International
+ * Journal for Numerical Methods in Fluids, 43(5), 555-575. The implementation
+ * is based on equations (70) and (79), which are adapted for the heat transfer
```
Comment (hepap) [inline include/solvers/vof_assemblers.h:165]:
```suggestion
 * is based on equations (70) and (79), which are adapted for the VOF
```

File: source/core/parameters.cc [hunk 5]
```diff
@@ -712,7 +712,15 @@ namespace Parameters
...
         Patterns::Bool(),
-        "Apply discontinuity-capturing directional dissipation stabilization term on heat transfer <true|false>");
+        "Apply discontinuity-capturing directional dissipation stabilization "
+        "term on heat transfer <true|false>");
+
+      prm.declare_entry(
+        "vof dcdd stabilization",
+        "false",
+        Patterns::Bool(),
+        "Apply discontinuity-capturing directional dissipation (dcdd) "
```
Comment (hepap) [inline source/core/parameters.cc:722]:
```suggestion
        "Apply Discontinuity-Capturing Directional Dissipation (DCDD) "
```

**Review by hepap — COMMENTED** [review]
Nicely done :) It's good to have the option now!

File: doc/source/parameters/cfd/stabilization.rst [hunk 6]
```diff
@@ -29,7 +31,9 @@ There are three choices of stabilization strategy:
 
 * ``stabilization=grad_div`` assembles a grad-div penalization term in the momentum equation to ensure mass conservation. This is not a stabilization method per-say and should not be used with elements that are not LBB stable. This stabilization should only be used with the grad-div block Navier-Stokes solver (``lethe-fluid-block``).
 
-* ``heat transfer stabilization``: Applies the discontinuity-capturing directional dissipation (DCDD) stabilization term on the heat transfer equation. For more information, see Tezduyar, T. E. (2003).
+* ``heat transfer stabilization``: Applies the discontinuity-capturing directional dissipation (DCDD) stabilization term on the heat transfer equation. For more information, see `Tezduyar, T. E. (2003) <https://doi.org/10.1002/fld.505>`_\.
+
+* ``vof dcdd stabilization``: Applies the DCDD stabilization term on the VOF equation. For more information, see `Tezduyar, T. E. (2003) <https://doi.org/10.1002/fld.505>`_\.
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/parameters/cfd/stabilization.rst:36]: Good idea!
I've added a link on "VOF equation".

File: source/core/parameters.cc (hunk 1 above)
Comment (AmishgaAlphonius) [reply source/core/parameters.cc:723]: That's true... Should I keep it enabled or disabled in tests ?

Comment (blaisb) [reply source/core/parameters.cc:723]: Keep it enabled in the tests so they don,t change. You can add one test where it is disabled though

**Review by hepap — APPROVED** [review]

**Review by blaisb — APPROVED** [review]

**Pull request merged**